src/provision.toit

```diff
@@ -34,17 +46,40 @@ class BLECharacteristic_:
         ble.BleUuid uuid
         --properties=PROPERTIES
         --permissions=PERMISSIONS
+        --read_timeout_ms=READ_TIMEOUT_MS
     characteristic.add_descriptor
         DESC_UUID
         PROPERTIES
         PERMISSIONS
         desc.to_byte_array
   
+    recv_task = task:: recv_task_run
+
+  recv_task_run:
+    characteristic.handle_read_request:
+      sem.down
+      sem.up
+      result
+  
   write data/ByteArray:
-    characteristic.write data
+    result = data
 
   read -> ByteArray:
     return characteristic.read
+  
+  is_encrypted -> bool:
+    return encrypted_
+
+  lock_host_read -> none:
+    sem.down
+  
+  unlock_host_read -> none:
+    sem.up
+  
```

Comment:
For example:
```suggestion
  mutex_do [block]:
    mutex_.do block.call
```

It would also be possible to expose the mutex and then use it directly.